packages/core/src/conversation/ConversationService/ConversationService.ts

```diff
@@ -716,7 +719,29 @@ export class ConversationService extends TypedEventEmitter<Events> {
   }
 
   private async handleMLSWelcomeMessageEvent(event: ConversationMLSWelcomeEvent) {
-    return this.mlsService.handleMLSWelcomeMessageEvent(event, this.apiClient.validatedClientId);
+    try {
+      return await this.mlsService.handleMLSWelcomeMessageEvent(event, this.apiClient.validatedClientId);
+    } catch (error) {
+      if (isCoreCryptoMLSOrphanWelcomeMessageError(error)) {
+        const {qualified_conversation: conversationId} = event;
+
+        // Note that we don't care about a subconversation here, as the welcome message is always for the parent conversation.
+        // Subconversations are always joined via external commit.
+
+        if (!conversationId) {
+          throw new Error('Qualified conversation id is missing in the event');
+        }
+
+        this.logger.info(
+          `Received an orphan welcome message, joining the conversation (${conversationId.id}) via external commit...`,
+        );
+
+        void queueConversationRejoin(conversationId.id, () => this.joinByExternalCommit(conversationId));
+        return null;
+      }
```

Comment:
I believe this logic should live in the `MLSService` rather than `ConversationService`. 

As much as possible the ConverstaionService should not be aware of MLS specific logic. 

Anyhow, not a blocker for this PR, can be done in a follow refactor PR :)